arbitrum-docs/for-devs/quickstart-solidity-hardhat.md

```diff
@@ -0,0 +1,491 @@
+
+
+Next, we'll deploy our smart contract to a network of real nodes: Arbitrum's Goerli testnet.
+
+### Deploy the smart contract to the Arbitrum Goerli testnet
+
+We were able to deploy to a local testnet for free because we were using [Hardhat's built-in Ethereum network emulator](https://hardhat.org/hardhat-network/docs/overview#hardhat-network). Because Arbitrum's Goerli testnet is powered by a real network of real nodes, we'll need to pay a small transaction fee to deploy our smart contract. This fee will be paid with the Arbitrum Goerli testnet's token, $AGOR.
+
+:::info $AGOR ISN'T REAL
+
+$AGOR isn't a canonical term, or a real token. It's just convenient shorthand for "Arbitrum Goerli testnet ETH" that we use to make this quickstart more readable.
+
+:::
+
+First, update the `hardhat.config.js` file to specify the private key of the test account that you'll use to deploy your smart contract (and pay the transaction fee):
+
+```javascript title="hardhat.config.js"
+// ...
+const GOERLI_TESTNET_PRIVATE_KEY = ""; // <- this should **not** begin with "0x"
+// ...
+accounts: [GOERLI_TESTNET_PRIVATE_KEY] // <- uncomment this line
+// ...
+```
+
+:::caution
+
+Note that we're adding a private key to a config file. This is **not** a best practice. In a real application, you should use a [signer](https://docs.ethers.io/v5/api/signer/) to sign transactions instead of handling private keys directly.
+
+:::
+
+Next, let's deposit some $AGOR into the wallet corresponding to the private key we added to `hardhat.config.js`. There are a few ways to acquire $AGOR:
+
+1. Use an L2 Goerli ETH faucet.
+2. Use an L1 Goerli ETH faucet and bridge your L1 Goerli ETH into Arbitrum L2 using [the Arbitrum bridge](https://bridge.arbitrum.io/).
+   - [goerlifaucet.com](https://goerlifaucet.com/) is a popular L1 Goerli ETH faucet.
+3. Ask someone nicely on [Discord](https://discord.gg/ZpZuw7p)[^6].
+
+Once you've acquired some $AGOR, you'll be able to deploy your smart contract to Arbitrum's Goerli testnet by issuing the following command:
+
+```bash
+yarn hardhat run scripts/deploy.js --network arbitrumGoerli
+```
+
+This tells hardhat to deploy the compiled smart contract through the RPC endpoint corresponding to `arbitrumGoerli` in `hardhat.config.js`. You should see the following output:
+
+
+```bash
+Cupcake vending machine deployed to 0xff825139321bd8fB8b720BfFC5b9EfDB7d6e9AB3
+```
+
+Congratulations! You've just deployed **business logic and data** to Arbitrum Goerli. This logic and data will find its way to Ethereum's L1 Goerli testnet, where it'll be hashed and stored in a Merkle tree and mirrored across all nodes in the Goerli network.
```

Comment:
`find its way to Ethereum's L1 Goerli testnet`
This might be confusing if the reader doesn't know how Arbitrum works. Maybe we can add a note with a link to how Arbitrum posts the information on L1?